src/js/components/stock-movement-wizard/combined-shipments/AddItemsPage.jsx

```diff
@@ -584,17 +602,57 @@ class AddItemsPage extends Component {
    * @public
    */
   saveAndTransitionToNextStep(formValues, lineItems) {
-    if (_.some(lineItems, item => item.inventoryItem
-      && item.expirationDate !== item.inventoryItem.expirationDate)) {
-      if (_.some(lineItems, item => item.inventoryItem && item.inventoryItem.quantity && item.inventoryItem.quantity !== '0')) {
-        this.confirmInventoryItemExpirationDateUpdate(() =>
-          this.saveRequisitionItemsAndTransitionToNextStep(formValues, lineItems));
-      } else {
-        this.saveRequisitionItemsAndTransitionToNextStep(formValues, lineItems);
-      }
-    } else {
-      this.saveRequisitionItemsAndTransitionToNextStep(formValues, lineItems);
+    if (_.some(lineItems, item =>
+      item?.expirationDate &&
+      item?.expirationDate !== item?.fetchedInventoryItem?.inventoryItem?.expirationDate &&
+      item?.fetchedInventoryItem?.quantityOnHand > 0)) {
+      this.confirmInventoryItemExpirationDateUpdate(() =>
+        this.saveRequisitionItemsAndTransitionToNextStep(formValues, lineItems));
+      return;
+    }
+    this.saveRequisitionItemsAndTransitionToNextStep(formValues, lineItems);
+  }
+
+  async fetchInventoryItem(values, rowIndex, isDebounced = false) {
+    if (isDebounced) {
+      this.debouncedInventoryItemFetch(values, rowIndex);
+      return;
     }
+
+    this.debouncedInventoryItemFetch.cancel();
+    const lotNumber = values?.lineItems[rowIndex]?.lotNumber;
+    const productCode = values?.lineItems[rowIndex]?.product?.productCode;
+
+    if (!lotNumber || !productCode) {
+      return;
+    }
+
+    const { data } = await ProductApi.getInventoryItem(productCode, lotNumber);
+    const mappedLineItems = this.state.values?.lineItems.map((lineItem, index) => {
+      if (rowIndex === index) {
+        return {
+          ...lineItem,
+          fetchedInventoryItem: data,
+        };
+      }
+      return lineItem;
+    });
+
+    this.setState(previousState => ({
+      ...previousState,
+      values: {
+        ...previousState.values,
+        lineItems: mappedLineItems,
+      },
+    }));
+  }
+
+  fetchMissingInventoryItems(values) {
+    values.lineItems.forEach((item, index) => {
+      if (!item.fetchedInventoryItem) {
+        this.fetchInventoryItem(values, index);
```

Comment:
this would mean, the amount of sent requests is equal to lineItems. 
Can't we make a batch request to fetch all missing at once?

Reply:
second question - what will the "missing inventory items" be? when will they be "missed"?
I assume that when you add a new one, it can't be missed, because you trigger the fetch immediately, don't you?
Unless by "missing inventory items" we mean items, that have been added there "some time ago" and we come back to the shipment to add some more?